package.json

```diff
@@ -48,6 +48,7 @@
   },
   "types": "./typings/index.d.ts",
   "dependencies": {
+    "abort-controller": "^3.0.0",
```

Comment:
Can I assume that you verified that there is no way around adding another dep if we want to process updates reliably around starting and stopping bots?

Reply:
Except for reimplementing it or dropping support for Node <15, there's no way.

Reply:
Yeah I see now that it's the way to go if you use `node-fetch` which we do.

Reply:
We could alternatively use more lightweight but less popular https://www.npmjs.com/package/node-abort-controller, I'll try it tomorrow

Reply:
`node-abort-controller` produces this:

```
src/core/network/polling.ts:30:11 - error TS2345: Argument of type 'AbortController' is not assignable to parameter of type 'CallApiOptions'.
  The types of 'signal.onabort' are incompatible between these types.        
    Type '((this: AbortSignal, ev: Event) => any) | null' is not assignable to type '((this: AbortSignal, event: any) => void) | null | undefined'.       
      Type '(this: AbortSignal, ev: Event) => any' is not assignable to type 
'(this: AbortSignal, event: any) => void'.
        The 'this' types of each signature are incompatible.
          Type 'import("D:/Wojciech/JS/telegraf/node_modules/@types/node-fetch/externals").AbortSignal' is not assignable to type 'AbortSignal'.
            Types of property 'onabort' are incompatible.
              Type '((this: AbortSignal, event: any) => void) | null | undefined' is not assignable to type '((this: AbortSignal, ev: Event) => any) | null'.
                Type 'undefined' is not assignable to type '((this: AbortSignal, ev: Event) => any) | null'.
```

Sticking with well-typed `abort-controller`.